docs/flow/modin/data_management/functions.rst

```diff
@@ -0,0 +1,127 @@
+Function Module Description
+"""""""""""""""""""""""""""
+
+Brief description
+'''''''''''''''''
+Most of the functions that are evaluated by `QueryCompiler` can be categorized into
+one of the patterns: Map, MapReduce, Binary functions, Fold functions, etc. Function
+module provides templates to easily build such types of functions. These templates
+are supposed to be used at the `QueryCompiler` level since each built function accepts
+and returns `QueryCompiler`.
+
+High-Level Module Overview
+''''''''''''''''''''''''''
+Each template represented with a class with the corresponding name and implements
+``register`` method, which takes functions to apply in an appropriate way and
+instantiate the related template. Functions that are passed to the ``register``
+will be executed under deserialized and preprocessed (depends on the template)
```

Comment:
What does it mean for functions to be executed under deserialized and preprocessed?